pyproject.toml

```diff
@@ -88,6 +88,11 @@ filterwarnings = [
     "ignore:jax.extend.mlir.dialects.mhlo is deprecated.*:DeprecationWarning",
     "ignore:jax.experimental.maps and .* are deprecated.*:DeprecationWarning",
     "ignore:The host_callback APIs are deprecated .*:DeprecationWarning",
+    "ignore:<Pjit:UserWarning",
```

Comment:
This is concerning to me: this ignore file applies to *all* JAX tests, and if `pjit` is raising a warning, we should catch it in our tests.

What warning are you filtering out here?

Reply:
The warning comes from the test suite's JSON reporting: 

```
.../array-api-tests/reporting.py:42: UserWarning: <PjitFunction of <function jax.numpy.greater at 0x7f16d9630c10>> (of type <class 'jaxlib.xla_extension.PjitFunction'>) is not JSON-serializable. Using the repr instead.
```

Reply:
Thanks - let's do something like this to make it more clear from the regex what exactly is being ignored (and to prevent potential false positives in the future).
```
ignore:.*is not JSON-serializable\. Using the repr instead\.
```

Reply:
Ruff doesn't seem happy with that in `pyproject.toml` , citing an "invalid escape sequence"😅. It also doesn't suppress the warnings when passed via `-W` flag for `pytest`, but not sure if the filters have slightly different behavior when parsed through the `.toml` file

Reply:
Removing the `\` characters should fix that I think